### DISCUSSION TITLE
return object after creation

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#186] [Q&A] [answered]
Hi,@wu-clan 
A common scenario is to return the object for the next step after creating the object.

However, an early commit refresh in the same transaction will return an exception.

```
async def create(self, db: AsyncSession, create: CreateUser) -> NoReturn:
     ......
    db.add(new_user)
    await db.commit()
    await db.refresh(new_user)
    return new_user
```

```
Can't operate on closed transaction inside context manager.  Please complete the context manager before emitting further commands.
```

**Comment by wu-clan** [comment]
sqlalchemy forbids this in asynchrony unless you put them in the same transaction, as in the paging interface, which transfers a db parameter instead of using the context manager

**Comment by downdawn** [comment, marked as answer]
solved

Use db.flush() to flush to the database, but do not commit, keep the transaction

**Reply by wu-clan** [reply]
Great, it is worth noting that flush just hangs and does not really write to the database.
